src/executor/execute.rs

```diff
@@ -48,11 +48,19 @@ pub fn execute<T: 'static + Debug, U: Store<T> + StoreMut<T>>(
         Prepared::Create(schema) => storage
             .insert_schema(&schema)
             .map(|(storage, _)| (storage, Payload::Create)),
-        Prepared::Insert(table_name, row) => {
-            let (storage, key) = storage.generate_id(&table_name)?;
-            let (storage, row) = storage.insert_data(&key, row)?;
-
-            Ok((storage, Payload::Insert(row)))
+        Prepared::Insert(table_name, rows) => {
+            match rows.into_iter().try_fold(
+                (storage, Vec::<Row>::new()),
+                |(storage, mut rows), row| {
```

Comment:
I recognized this `mut` and I also recently stuck at collecting vector using immutable api.
Even `im_rc::Vector` doesn't have sufficient immutable apis, recently I'm thinking about making immutable wrapper for `std Vector` and `std HashMap`.
Anyway for this issue, let's take a different approach.

At the beginning when I decided the result type of `INSERT` as `Row`, it wasn't because of something special reason. I just chose it. And the situation which only requires to return a single `Row`, it was ok, just for a single row.
But it looks inefficient to return all successfully inserted row.
Like `DELETE` and `UPDATE` queries, it would be just enough to return number of affected rows, not full row data.

Let's remove it at this point, but using the different issue.
https://github.com/gluesql/gluesql/issues/70

For this issue, I'll merge PR with that `mut` use.
You can remove it working on issue #70, can you please also take a look that issue?

Reply:
Sure, I'll get on it ASAP